data/transactions/application.go

```diff
@@ -112,10 +112,10 @@ func (ac ApplicationCallTxnFields) Empty() bool {
 	if ac.GlobalStateSchema != (basics.StateSchema{}) {
 		return false
 	}
-	if ac.ApprovalProgram != "" {
+	if len(ac.ApprovalProgram) != 0 {
```

Comment:
I think in this case we want `!= nil`, same below.

Reply:
done